File: web/src/api/approval.ts
import { apiFetch } from './client'

export interface ApprovalItem {
  id: string
  created_at: string
  source: string
  task_name: string
  action_type: string
  risk_level: string
  reason: string
  prompt: string
  full_output: string
  status: 'pending' | 'accepted' | 'rejected' | 'expired'
  approved_at?: string
}

export function fetchPendingApprovals(): Promise<ApprovalItem[]> {
  return apiFetch<ApprovalItem[]>('/approvals/pending')
}

export function fetchAllApprovals(): Promise<ApprovalItem[]> {
  return apiFetch<ApprovalItem[]>('/approvals')
}

export function decideApproval(id: string, action: 'accept' | 'reject' | 'skip' | boolean): Promise<void> {
  const body = typeof action === 'boolean'
    ? { accept: action }
    : { action }
  return apiFetch<void>(`/approvals/${id}/decide`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(body),
  })
}

File: internal/model/approval.go
package model

import "time"

// ApprovalStatus for pending action confirmation.
type ApprovalStatus string

const (
	ApprovalPending  ApprovalStatus = "pending"
	ApprovalAccepted ApprovalStatus = "accepted"
	ApprovalRejected ApprovalStatus = "rejected"
	ApprovalExpired  ApprovalStatus = "expired"
)

// ApprovalItem represents a single pending approval request from the agent.
type ApprovalItem struct {
	ID          string         `json:"id"`
	CreatedAt   time.Time      `json:"created_at"`
	Source      string         `json:"source"`      // "cron" / "chat"
	TaskName    string         `json:"task_name"`    // cron job name or conversation ID
	ActionType  string         `json:"action_type"`  // brief description of the action
	RiskLevel   string         `json:"risk_level"`   // 低/中/高
	Reason      string         `json:"reason"`       // why approval is needed
	Prompt      string         `json:"prompt"`       // the original prompt that triggered this
	FullOutput  string         `json:"full_output"`  // complete agent output for review
	Status      ApprovalStatus `json:"status"`
	ApprovedAt  *time.Time     `json:"approved_at,omitempty"`
}
